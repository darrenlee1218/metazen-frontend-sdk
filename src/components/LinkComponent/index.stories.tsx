import React from 'react';
import { ComponentStory } from '@storybook/react';

import LinkComponent from '.';

export default {
  title: 'common-components/stateless/atoms/LinkComponent',
  component: LinkComponent,
  argTypes: {},
};

const Template: ComponentStory<typeof LinkComponent> = (args) => <LinkComponent {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Link',
};
