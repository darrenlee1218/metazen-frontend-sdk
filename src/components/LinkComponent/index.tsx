import React from 'react';
import Link, { LinkProps } from '@mui/material/Link';

const LinkComponent: React.FC<LinkProps> = (props) => (
  <Link sx={{ cursor: 'pointer', pb: '8px' }} underline="hover" color="text.primary" {...props}>
    {props.children}
  </Link>
);

export default LinkComponent;
