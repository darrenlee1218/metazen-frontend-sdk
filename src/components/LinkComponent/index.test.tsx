import React from 'react';
import TestRenderer from 'react-test-renderer';

import LinkComponent from './index';

describe('Test snapshot LinkComponent components', () => {
  it('should be render correctly', () => {
    const tree = TestRenderer.create(<LinkComponent>Link</LinkComponent>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
