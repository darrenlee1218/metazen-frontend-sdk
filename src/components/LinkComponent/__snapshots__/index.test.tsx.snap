// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot LinkComponent components should be render correctly 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-kpicih-MuiTypography-root-MuiLink-root"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Link
</a>
`;
